### PR TITLE
Implement fourmolu everywhere

### PR DESCRIPTION
**Pull request by lf-**
This makes it easier to use fourmolu as a format-on-save code-fixer.

### CODE REVIEW

File: tests/TestImport.hs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  )
+where
+
+import ClassyPrelude hiding (link)
+import Control.Monad.Fail
+import Data.Aeson (FromJSON (..), ToJSON (..), object, withObject, (.:), (.:?), (.=))
+import Data.Aeson.TH (deriveJSON)
+import Data.Maybe (fromJust)
+import Data.String.Conversions (cs)
+import Test.Hspec
```
Comment (lf-) [inline tests/TestImport.hs:18]: This is the singular manually written change in the testsuite along with making the tests import it

**Review by lf- — COMMENTED** [review]
Most of this is automated changes. GitHub seems to have largely conveniently reordered everything manually changed and actually relevant to the top of the diff

File: .github/workflows/haskell-ci.yml [hunk 2]
```diff
@@ -14,8 +14,12 @@
 #
 name: Haskell-CI
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - master
```
Comment (friedbrice) [inline .github/workflows/haskell-ci.yml:19]: should we just disable pushing to master?

Comment (lf-) [reply .github/workflows/haskell-ci.yml:19]: We already did; the goal of this is to CI check the result of merging to master (which is not necessarily the CI check of any given PR since PRs may be based on older master).

File: Setup.hs [hunk 3]
```diff
@@ -1,2 +0,0 @@
-import           Distribution.Simple
-main = defaultMain
```
Comment (friedbrice) [inline Setup.hs:2]: I have no idea why this is still included with project templates -_-

Comment (lf-) [reply Setup.hs:2]: me neither, so i deleted it

File: fourmolu.yaml [hunk 4]
```diff
@@ -6,6 +6,8 @@ diff-friendly-import-export: false # 'false' uses Ormolu-style lists
 respectful: false # don't be too opinionated about newlines etc.
 haddock-style: single-line # 'signle-line' or 'multi-line', i.e., '--' vs. '{-'
 newlines-between-decls: 1 # number of newlines between top-level declarations
+ghc-opt:
+  -XImportQualifiedPost
```
Comment (friedbrice) [inline fourmolu.yaml]: Why do we need this in `fourmolu.yaml` _and_ in `flake.nix`? IIUC, `flake.nix` is going to set it has an environment variable, so having it in both places is redundant? Of course I might just not understand correctly what `flake.nix` is going ^_^;

Comment (lf-) [reply fourmolu.yaml]: oh that's a mistake! fourmolu actually doesn't look at this field in the config file; it is supposed to look at the cabal files instead. *however*, for some reason, it doesn't do so in the pre-commit hook mode, which smells like a bug.

Comment (lf-) [reply fourmolu.yaml]: Filed the bug https://github.com/cachix/pre-commit-hooks.nix/issues/171

File: src/Web/Slack/Conversation.hs [hunk 5]
```diff
@@ -181,69 +159,62 @@ data GroupConversation =
...
-      is <- (o .: key) <|> empty
-      if is
-        then con <$> parseJSON (Object o)
-        else empty
-
+        "parsing a Conversation failed: neither channel, group, nor im, "
+        (typeMismatch "Conversation" (Object o))
+    where
+      parseWhen key con o = do
+        is <- (o .: key)
```
Comment (friedbrice) [inline src/Web/Slack/Conversation.hs:214]: This is really the only real question I have on this PR. What effect does removing the `<|> empty` have?

So, I had to think about this. Either `o` has the key or it doesn't have the key.

If `o` has the key, then `o .: key` and `o .: key <|> empty` are the same, they both succeed and bind the value there to a `is :: Bool`. The subsequent code is the same in both versions, so no difference.

If `o` doesn't have the key, then `o .: key` short circuits the parser and possibly gives some context, whereas `o .: key <|> empty` short circuits the parser and provides no context. So the difference is that the new version `o .: key`, might have some additional context. The consuming code hasn't changed in this PR, so the new context, if any, isn't used in an essential way: it isn't branched on.

In summary, it looks like the new version, omitting `<|> empty`, provides at least as much information as the original version, but no new information is used downstream.

**Review by friedbrice — APPROVED** [review]

File: src/Web/Slack/Conversation.hs (hunk 5 above)
Comment (lf-) [reply src/Web/Slack/Conversation.hs:214]: That was because hlint deleted it with the mwb rules. So I expect it should be fine.
